src/main/java/com/yahoo/bullet/storm/QuerySpout.java

```diff
@@ -23,36 +30,51 @@
     public static final String QUERY_FIELD = "query";
     public static final String METADATA_FIELD = "metadata";
 
-    private PubSub pubSub;
+    private BulletStormConfig config;
     private Subscriber subscriber;
     private SpoutOutputCollector collector;
 
+    /** Exposed for testing only. */
+    @Getter(AccessLevel.PACKAGE)
+    private PubSub pubSub;
+
     /**
-     * Creates a QuerySpout and passes in a {@link PubSub}.
+     * Creates a QuerySpout with a passed in {@link BulletStormConfig}.
      *
-     * @param pubSub PubSub to get a {@link Subscriber} from
+     * @param config The BulletStormConfig to create the PubSub from.
      */
-    public QuerySpout(PubSub pubSub) {
-        this.pubSub = pubSub;
+    public QuerySpout(BulletStormConfig config) {
+        this.config = config;
     }
 
     @Override
     public void open(Map conf, TopologyContext context, SpoutOutputCollector collector) {
+        // Add the Storm Config and the context as is, in case any PubSubs need it.
+        config.set(BulletStormConfig.STORM_CONFIG, conf);
+        config.set(BulletStormConfig.STORM_CONTEXT, context);
+
         this.collector = collector;
-        this.subscriber = pubSub.getSubscriber();
+        try {
+            pubSub = PubSub.from(config);
+            subscriber = pubSub.getSubscriber();
+            log.info("Setup PubSub: {} with Subscriber: {}", pubSub, subscriber);
+        } catch (PubSubException e) {
+            throw new RuntimeException("Cannot create PubSub instance or a Subscriber for it.", e);
+        }
     }
 
     @Override
     public void nextTuple() {
         PubSubMessage message = null;
         try {
             message = subscriber.receive();
-        } catch (PubSubException e) {
+        } catch (Exception e) {
```

Comment:
What else exception does it throw? I checked the interface definition, only PubSubException thrown.

Reply:
To catch RuntimeExceptions. Basically if anything bad happens for any reason, we don't want to restart the spout (ex. bad JSON in PubSubMessage throws a GSON RuntimeException)